satpy/readers/hdfeos_mod35.py

```diff
@@ -0,0 +1,434 @@
+
+import dask.array as da
+import xarray.ufuncs as xu
+import xarray as xr
+from satpy import CHUNK_SIZE
+from satpy.dataset import DatasetID
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy.readers.hdf4_utils import from_sds
+
+logger = logging.getLogger(__name__)
+
+import time
+
+class HDFEOSFileReader(BaseFileHandler):
+
+    def __init__(self, filename, filename_info, filetype_info):
+        super(HDFEOSFileReader, self).__init__(filename, filename_info, filetype_info)
+        self.filename = filename
+        try:
+            self.sd = SD(str(self.filename), SDC.READ)
+        except HDF4Error as err:
+            raise ValueError("Could not load data from " + str(self.filename)
+                             + ": " + str(err))
+        self.metadata = self.read_mda(self.sd.attributes()['CoreMetadata.0'])
+        self.metadata.update(self.read_mda(
+            self.sd.attributes()['StructMetadata.0']))
+        self.metadata.update(self.read_mda(
+            self.sd.attributes()['ArchiveMetadata.0']))
+
+    @property
+    def start_time(self):
+        date = (self.metadata['INVENTORYMETADATA']['RANGEDATETIME']['RANGEBEGINNINGDATE']['VALUE'] + ' ' +
+                self.metadata['INVENTORYMETADATA']['RANGEDATETIME']['RANGEBEGINNINGTIME']['VALUE'])
+        return datetime.strptime(date, '%Y-%m-%d %H:%M:%S.%f')
+
+    @property
+    def end_time(self):
+        date = (self.metadata['INVENTORYMETADATA']['RANGEDATETIME']['RANGEENDINGDATE']['VALUE'] + ' ' +
+                self.metadata['INVENTORYMETADATA']['RANGEDATETIME']['RANGEENDINGTIME']['VALUE'])
+        return datetime.strptime(date, '%Y-%m-%d %H:%M:%S.%f')
+
+    def read_mda(self, attribute):
+        lines = attribute.split('\n')
+        mda = {}
+        current_dict = mda
+        path = []
+        for line in lines:
+            if "gdas1" in line:
+                continue
+            if not line:
+                continue
+            if line == 'END':
+                break
+            keyval = line.split('=')
+            if len(keyval) <= 1:
+                continue
+            key, val =  keyval
+            key = key.strip()
+            val = val.strip()
+            try:
+                val = eval(val)
+            except NameError:
+                pass
+            if key in ['GROUP', 'OBJECT']:
+                new_dict = {}
+                path.append(val)
+                current_dict[val] = new_dict
+                current_dict = new_dict
+            elif key in ['END_GROUP', 'END_OBJECT']:
+                if val != path[-1]:
+                    raise SyntaxError
+                path = path[:-1]
+                current_dict = mda
+                for item in path:
+                    current_dict = current_dict[item]
+            elif key in ['CLASS', 'NUM_VAL']:
+                pass
+            else:
+                current_dict[key] = val
+        return mda
+
+
+class HDFEOSGeoReader(HDFEOSFileReader):
```

Comment:
Same here, it feels like this class has a lot in common with hdfeos l1b reader, can we have a common baseclass ?